### ISSUE TITLE
Should `host_metrics` uptime be a counter?

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#7362]
Asked by a [user in discord](https://discord.com/channels/742820443487993987/746070591097798688/839715852420644884).

We currently publish `uptime` as a gauge but it seems like it could be a counter (`uptime_secs_total`) since it never decreases unless the instance is rebooted.

@bruceg I'm curious if you have thoughts or if there is a reason that it is a gauge that I'm not aware of.

**Comment by bruceg** [comment]
This was likely due to prior requirement that counters be incremental, leading to the difficulty in maintaining the resulting absolute value, whereas gauges are typically absolute. However, as we now have the capability of doing absolute values for counters #5874, this is a possibility.

Thinking further, though, some of the metrics _sinks_ expose all counters as incremental, and all gauges as absolutes (this in fact was the "standard" behavior for all sinks using the metrics batch buffer). This has the implication that exposing this value as a counter would result in a vector restart not being visible in the metric stream as a discontinuity in the `uptime` value, since vector would keep exporting a constant counter increment. AFAIK this is pretty much _the_ important reason for monitoring this value, so an incremental counter eliminates this possibility.

So, while I am open to alternate reasoning, I think this needs to stay as a gauge.

**Comment by jszwedko** [comment]
@bruceg thanks for the thoughts! I think I'm failing to see how this is any different than the existing absolute counters that `host_metrics` publishes like `network_transmit_packets_total`. I don't think we should be publishing incremental counters for those since the source of the metric data is absolute. I think the same applies to the `uptime` metric here which is fetched from the host itself.

**Comment by bruceg** [comment]
I'd say I/O counters like you describe are desirable to maintain across restarts since the traffic from one run of vector is not really distinct from that of another run, hence the incremental behavior, whereas `uptime` is specifically there to catch restarts.

**Comment by jszwedko** [comment]
@bruceg I think we may be talking about different metrics? The `uptime` metric in `host_metrics` refers to the host uptime, not Vector's uptime. I believe Vector's uptime is reported separately via `internal_metrics.

**Comment by bruceg** [comment]
Ah, yes, I was talking about vector's uptime, but I think the same point stands for the host uptime as well. Uptime is (again AFAIK) tracked to see how long _this instance_ (either host or run of vector) has been running, which cannot be properly exposed by an incremental value, while I/O counters are tracked to monitor rates, for which an incremental is most appropriate.

I agree this is technically a counter for the instance, since it will never decrease, but it needs to remain a gauge in practice since we want to catch those edges on restarting.

**Comment by jszwedko** [comment]
Right, it needs to be exposed as an absolute counter. I think that the I/O counters are same though, no? What if I report, say `50` for `network_transmit_packets_total`. I stop vector and then start it back up some time later when `network_transmit_packets_total` would be 100. An incremental counter saying `0` would not be correct. We'd want to publish the absolute value of `100`. Those values are fetched from `/proc/net` rather than recorded by vector while it is up.
